tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2196,6 +2196,52 @@ def visit_functiondef(self,node):
         except:
             pass
 
+class NoLegacyAzureCoreHttpResponseImport(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that we aren't importing azure.core.pipeline.transport.HttpResponse outside of Azure Core."""
+    name = "no-legacy-azure-core-http-response-import"
+    priority = -1
+    msgs = {
+        "C4756": (
+            "Do not import HttpResponse from azure.core.pipeline.transport outside of Azure Core.",
+            "no-legacy-azure-core-http-response-import",
+            "Do not import HttpResponse from azure.core.pipeline.transport outside of Azure Core. You can import HttpResponse from azure.core.rest instead."
+        ),
+    }
+
+    AZURE_CORE_NAME = "azure.core"
+    AZURE_CORE_TRANSPORT_NAME = "azure.core.pipeline.transport"
+    RESPONSE_CLASSES = ["HttpResponse", "AsyncHttpResponse"]
+
+    def visit_importfrom(self, node):
+        """Check that we aren't importing from azure.core.pipeline.transport import HttpResponse."""
+        if node.root().name.startswith(self.AZURE_CORE_NAME): 
+            return
+        if node.modname == self.AZURE_CORE_TRANSPORT_NAME: 
+            for name, _ in node.names:
+                if name in self.RESPONSE_CLASSES:
+                    self.add_message(
+                        msgid=f"no-legacy-azure-core-http-response-import",
+                        node=node,
+                        confidence=None,
+                    )
+    
+    def visit_import(self, node):
+        """Check that we aren't importing import azure.core.pipeline.transport.HttpResponse."""
```

Comment:
I think we might have to include 2 separate checks here, but let me know if I'm wrong. In my head, we do one check in the file for a `import azure.core.pipeline.transport` and then we have to check/visit the code in the file for any usage `azure.core.pipeline.transport.HttpResponse`.

Reply:
I tried a global search in the python repo currently and it doesn't look like there are any instances of just import azure.core.pipeline.transport, so I don't know how often this would be used but I can add it! Maybe it will be more likely to pop up when typing gets added places?

Finding every individual usage of httpResponse (without digging into astroid yet -- there could be a faster way) would require iteration through every funcDef and then within that funcDef check every assign node when we see a file that has `import azure.core.pipeline.transport` at the top 

i.e something like : response = azure.core.pipeline.transport.HttpResponse(.....

Reply:
FYI I'm fine with just checking the `from x import y` style import, but wanted to double-check since I think the issue you created had said something like check *all* ways of importing it. If you think this will be a big perf hit for a situation unlikely to happen then maybe we don't need it

Reply:
So should we remove the visit_import function?